geo-data-server/src/main/java/org/gridsuite/geodata/server/GeoDataService.java

```diff
@@ -236,6 +236,49 @@ void saveLines(List<LineGeoData> linesGeoData) {
         lineRepository.saveAll(linesEntities);
     }
 
+    boolean emptyOrEquals(String emptyable, String s) {
+        return emptyable.isEmpty() || s.equals(emptyable);
+    }
+
+    private LineGeoData getLineGeoDataWithEndSubstations(Map<String, LineGeoData> linesGeoDataDb, Map<String, SubstationGeoData> substationGeoDataDb, Line line) {
+        LineGeoData geoData = linesGeoDataDb.get(line.getId());
+        Substation sub1 = line.getTerminal1().getVoltageLevel().getSubstation();
+        Substation sub2 = line.getTerminal2().getVoltageLevel().getSubstation();
+        Coordinate substation1Coordinate = substationGeoDataDb.get(sub1.getId()).getCoordinate();
+        Coordinate substation2Coordinate = substationGeoDataDb.get(sub2.getId()).getCoordinate();
+
+        if (geoData == null || geoData.getCoordinates().isEmpty() || (geoData.getSubstationStart().isEmpty() && geoData.getSubstationEnd().isEmpty())) {
+            return new LineGeoData(line.getId(), sub1.getNullableCountry(), sub2.getNullableCountry(), sub1.getId(), sub2.getId(),
+                List.of(substation1Coordinate, substation2Coordinate));
+        }
+        boolean reverse = false;
+        if (emptyOrEquals(geoData.getSubstationStart(),  sub2.getId()) && emptyOrEquals(geoData.getSubstationEnd(), sub1.getId())) {
+            reverse = true;
+        } else if (!(emptyOrEquals(geoData.getSubstationStart(), sub1.getId()) && emptyOrEquals(geoData.getSubstationEnd(), sub2.getId()))) {
```

Comment:
why only test for a different "origin end" ? Why not also test if the other end is wrong ?